tracer/src/Datadog.Tracer.Native/cor_profiler.cpp

```diff
@@ -134,14 +128,32 @@ HRESULT STDMETHODCALLTYPE CorProfiler::Initialize(IUnknown* cor_profiler_info_un
         return CORPROF_E_PROFILER_CANCEL_ACTIVATION;
     }
 
-    const auto& exclude_process_names = shared::GetEnvironmentValues(environment::exclude_process_names);
-
-    // attach profiler only if this process's name is NOT on the list
-    if (!exclude_process_names.empty() && shared::Contains(exclude_process_names, process_name))
+    // if we were on the explicit include list, don't check the block list
+    if (include_process_names.empty())
     {
-        Logger::Info("DATADOG TRACER DIAGNOSTICS - Profiler disabled: ", process_name, " found in ",
-                     environment::exclude_process_names, ".");
-        return CORPROF_E_PROFILER_CANCEL_ACTIVATION;
+        const auto& exclude_process_names = shared::GetEnvironmentValues(environment::exclude_process_names);
+        if (!exclude_process_names.empty())
+        {
+            // attach profiler only if this process's name is NOT on the provided list
```

Comment:
We've had a "Profiler" product for a while so this could be confusing. We should stop using "profiler" when we mean the "native tracer library." Or whatever better name you can come up with 😅

Reply:
If only we weren't in the `cor_profiler` file... :stare:

Reply:
I usually use "clrprofiler"